src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2310,6 +2323,33 @@ def reset_index(
 
         return SnowflakeQueryCompiler(internal_frame)
 
+    def reindex(
```

Comment:
why we add this?

Reply:
We use it [here](https://github.com/snowflakedb/snowpark-python/blob/f780dbab7ac7ed201135421aedcdab3ff914601d/src/snowflake/snowpark/modin/pandas/dataframe.py#L230)

Reply:
This one actually call default to pandas internally, now we don't do this in qc. Can you do this instead?

```
new_qc = new_qc.reindex(axis=0, labels=index)

=>

new_qc = new_qc.super().reindex(axis=0, labels=try_convert_index_to_native(labels))
```

Reply:
or can you use implemented method, e.g., set_index?

Reply:
`new_qc = new_qc.reindex(axis=0, labels=try_convert_index_to_native(index))`
I initially wanted to do this but I thought we wanted to minimize front end changes, thats why I created a method in SnowflakeQueryCompiler to do the conversion. Do you want me to just do it like this?

Reply:
if you can implement a real reindex method, then that's good but it is not. 
Now your index.py is under plugin so we don't need to worry about the frond end changes.

Reply:
Ok I removed from SnowflakeQueryCompiler